client/rest-high-level/src/main/java/org/elasticsearch/client/indices/rollover/RolloverRequest.java

```diff
@@ -116,6 +118,19 @@ public RolloverRequest addMaxIndexSizeCondition(ByteSizeValue size) {
         this.conditions.put(maxSizeCondition.name(), maxSizeCondition);
         return this;
     }
+
+    /**
+     * Adds a size-based condition to check if the size of the largest primary shard is at least <code>size</code>.
+     */
+    public RolloverRequest addMaxIndexSinglePrimarySizeCondition(ByteSizeValue size) {
```

Comment:
I think this can remove the "Index" part, so it can be `addMaxSinglePrimarySizeCondition`

Reply:
Ehhhhh, I'm with you on disliking the name, but the pattern for the other conditions ends up with an `Index` in the middle -- is it worth breaking the apparent convention for these?

Reply:
I don't think it's a convention as much as the others make "English" sense without the `Index` in them (it could be argued that "Size" would be fine without it, but that one will hopefully eventually be deprecated). I'm not super concerned, so if you'd rather keep the `Index` that's fine with me.

Reply:
Renamed in d00e2287e4b9de6ae5fb1b900847439b74ca0242